### PR TITLE
Add Working Group Life Cycle

### PR DESCRIPTION
**Pull request by dcmiddle**
The initial draft of the working group life cycle is based on
https://github.com/ossf/tac/issues/13 and TAC discussions leading up to
the 2020-12-01 meeting.

It is intended to satisfy the Open SSF Charter item 6.e.iii:
[The TAC will be responsible for: ...]
"iii) creating, maintaining and amending project lifecycle states, review
procedures and processes;"

Signed-off-by: Dan Middleton <dan.middleton@intel.com>

### CODE REVIEW

File: working-group-lifecycle.md
Comment (dlorenc) [inline working-group-lifecycle.md:46]: 20 times is basically an entire year - I'd be in favor of dropping this to 10.

File: working-group-lifecycle.md
Comment (dlorenc) [inline working-group-lifecycle.md:53]: This one still doesn't make sense to me as a working group lead. What does a WG get in return for the extra overhead of producing annual goals and metrics?

I'm guessing this would get dropped over time and not strictly enforced, which is maybe not a bad thing.

**Review by dlorenc — COMMENTED** [review]
Thanks Dan! I think this accurately captures where we left off yesterday.

File: working-group-lifecycle.md
Comment (jenniferfernick) [reply working-group-lifecycle.md:46]: I'm also in favour of it being a smaller number (such as 10). 

Could we modify this to be time or objective based, to reward progress? For example, we could try something like, "Have met at least 10 times and/or have published at least one major tool, paper, or library"

File: working-group-lifecycle.md
Comment (jenniferfernick) [inline working-group-lifecycle.md:23]: I would love if we could provide some "seed" resources to incubating projects and lay out a guide for how to access each one. This is arguably the time when they'll need our support the most. 

What subset of the following could we offer without too much admin overhead?  
"Have access to community resources (Zooms, YouTube channels, GitHub, Slack channels, etc.)"

**Comment by estesp** [comment]
Minor point on formatting; 2 of the 6 headings for state changes end with a colon; 4 do not. I don't have a strong preference either way, but should be consistent.

Question: for the proposed numeric checks (5 people, 3 orgs; or 10 people involved, etc.) do we have any data on current WGs and where they sit against those criteria? Given I personally haven't been involved in the WGs to date, I don't have a sense for where that bar sits with respect to existing groups. Given that, it's hard to assess against my gut feel that these might be higher than expected. For example, there are some vibrant OSS projects with 5 or fewer people actively contributing.

Corollary: do we want to say anything about "judgement calls" for the TAC decision process and that the suggested items are effectively markers but not hard and fast rules?

**Comment by dcmiddle** [comment]
See latest commit (and commit message) for reducing the meeting count and attendance numbers.
For rationale of separate active and incubating stages see the last paragraph of the intro, i.e., there's a checkpoint to ensure incubation completed correctly.
It's not a big deal either way but it's probably beneficial to have a checkpoint.

File: working-group-lifecycle.md
Comment (dlorenc) [reply working-group-lifecycle.md:23]: +1! What do you have in mind for these resources?

**Comment by dlorenc** [comment]
Thanks for the updates Dan! I'm happy with this as a starting point, we can always iterate as we go.

One thing that got dropped somewhere between the initial discussion in #13 and now appears to be any kind of explanation on what it means to be a working group in either of the levels. From a practical perspective - what can a working group do, other than run meetings? This should cover stuff like:

- Can an active working group publish a white paper without tac approval? 
- What about an incubating one?
- Can a working group create repositories? When does it make sense to put these "in a WG" vs. treat them as new, standalone things?
- What about fundraising/soliciting donations?
- Where should repos/code "live"? This org or somewhere else?

I had some thoughts higher up in #13, but I think it's probably ok to punt on this for now. I do think we probably want to revisit that aspect before starting any reviews of the existing working groups though. For the time being the TAC has been erring on the side of minimal oversight/maximum autonomy which is great with me. It's hard to design governance in the abstract, so observing what the WGs are doing now will make this easier later. We're still small and it's still relatively easy to keep an eye on things.

Any thoughts from others?

**Comment by JonZeolla** [comment]
As someone new to the OSSF and looking to get involved in some WGs I did find it helpful that there was a maintained index [under the tac](https://github.com/ossf/tac#technical-initiatives)  but it's possible not everyone will look at the tac README.  To keep discoverablility high I would suggest keeping all the repos under the OSSF org, maintain a working group topic or naming standard, and then maybe linking to a filtered query from the openssf.org website

File: working-group-lifecycle.md
Comment (david-a-wheeler) [reply working-group-lifecycle.md:23]: We should probably add one additional state, `proposed`, for a proposed WG that hasn't entered the `incubating` state.

**Review by dlorenc — APPROVED** [review]

**Review by estesp — APPROVED** [review]
LGTM

**Comment by christo4ferris** [comment]
This is fine for a working group, I suppose, though I am still unclear about things such as who has commit privs for the created repository or how the WG intends to manage that over time. However, we now have a mix of projects (repos) and working groups, and no clear distinction between them. We also have no guidance as to what the minimum expectations are for the repository (things such as a README.md, CONTRIBUTING.md, LICENSE, MAINTAINERS.md, etc). I also don't understand how number of meetings and diversity of representation make a WG "active". The bar seems awfully low. How about ensuring that their meetings are advertised on the community calendar, that their agenda and minutes are posted along with recordings, etc. How about assessing whether the WG is actually producing something of value?

**Comment by christo4ferris** [comment]
> Thanks for the updates Dan! I'm happy with this as a starting point, we can always iterate as we go.
> 
> One thing that got dropped somewhere between the initial discussion in #13 and now appears to be any kind of explanation on what it means to be a working group in either of the levels. From a practical perspective - what can a working group do, other than run meetings? This should cover stuff like:
> 
>     * Can an active working group publish a white paper without tac approval?
> 
>     * What about an incubating one?
> 
>     * Can a working group create repositories? When does it make sense to put these "in a WG" vs. treat them as new, standalone things?
> 
>     * What about fundraising/soliciting donations?
> 
>     * Where should repos/code "live"? This org or somewhere else?
> 
> 
> I had some thoughts higher up in #13, but I think it's probably ok to punt on this for now. I do think we probably want to revisit that aspect before starting any reviews of the existing working groups though. For the time being the TAC has been erring on the side of minimal oversight/maximum autonomy which is great with me. It's hard to design governance in the abstract, so observing what the WGs are doing now will make this easier later. We're still small and it's still relatively easy to keep an eye on things.
> 
> Any thoughts from others?

I really don't think it wise to punt on these items. They will only result in confusion. We have a number of incubating WGs now. Can they create repositories without oversight? How are they managed? have we policies for best practices for how a repository is structured and governed? Way too many open switches.

**Comment by estesp** [comment]
@christo4ferris the disconnection between meeting minutes and GH issues/activity is at fault here. The TAC acknowledged all those items as needing further discussion, but getting a framework moving forward was deemed an important first step while we take the time to dig into more detail on the "how"/"what"/etc. of WG lifecycle activities. Given we are doing this as a GH markdown guide, iteration is relatively easy and PRs can continue to build on this initial framework.

**Comment by dcmiddle** [comment]
See https://github.com/ossf/tac/issues/44 for what Active gets you.

**Review by rhaning — APPROVED** [review]

**Review by estesp — APPROVED** [review]
LGTM

File: working-group-lifecycle.md
Comment (mayakacz) [inline working-group-lifecycle.md:3]:
```suggestion

## Starting a new working group

Are you interested in starting a new working group? Contributors interested in starting a new working group are encouraged to first review the existing
```

File: working-group-lifecycle.md
Comment (mayakacz) [inline working-group-lifecycle.md:5]:
```suggestion
cover the new topic. If this is the case, please bring your time and energy into that existing working
```

File: working-group-lifecycle.md
Comment (mayakacz) [inline working-group-lifecycle.md:9]:
```suggestion
meetings, mailing lists, and/or Slack channels to solicit support for a new working group.
```

File: working-group-lifecycle.md
Comment (mayakacz) [inline working-group-lifecycle.md:14]: Feels like this should be at the top maybe?

File: working-group-lifecycle.md
Comment (mayakacz) [reply working-group-lifecycle.md:23]: > We should probably add one additional state, `proposed`, for a proposed WG that hasn't entered the `incubating` state.

I think I'm fine with just going straight to `incubating`.

> What subset of the following could we offer without too much admin overhead?
"Have access to community resources (Zooms, YouTube channels, GitHub, Slack channels, etc.)"

Now that that's included in the preamble, I don't think we need to include it here. Alternatively, something like "Can use community resources to organize, but do not have dedicated channels"

File: working-group-lifecycle.md
Comment (mayakacz) [reply working-group-lifecycle.md:57]: I think I originally suggested this because we wanted WGs to set goals and have a vision of what they're working on. But yeah, I agree with you that this is not necessarily the right thing to aim for - I also don't think there are necessarily expectations around the goals being met.
Maybe there's nothing 'additional' to do once you're active, as long as you keep doing the above. I'd be OK to remove.

**Comment by dlorenc** [comment]
OK! This is approved now with 4. Ready to merge.

**Pull request merged**
